fx quant: add types to observed_module.py

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #49717 fx quant: hook up ConvTranspose{n}d
* #49688 fx quant: improve types on convert
* #49628 fx quant: do not observe bias on F.linear
* #49623 fx quant: do not observe bias on F.conv
* #49616 fx quant: fix types on _find_quants
* **#49607 fx quant: add types to observed_module.py**
* #49606 fx quant: types for fusion_patterns.py
* #49549 unbreak mypy torch/quantization

Summary:

Readability

Test Plan:

```
mypy torch/quantization
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25643895](https://our.internmc.facebook.com/intern/diff/D25643895)

vkuzo committed:
fx quant: add types to observed_module.py

Summary:

Readability

Test Plan:

```
mypy torch/quantization
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/quantization/fx/observed_module.py:
```diff
@@ -26,10 +28,10 @@ def __deepcopy__(self, memo):
     return ObservedGraphModule(module, module.graph)
 
-def is_observed_module(module):
+def is_observed_module(module: Any) -> bool:
```
can this be GraphModule as well?

vkuzo committed:
Update on "fx quant: add types to observed_module.py"


Summary:

Readability

Test Plan:

```
mypy torch/quantization
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25643895](https://our.internmc.facebook.com/intern/diff/D25643895)

[ghstack-poisoned]
